File: skypydb-js/src/security/mixins/validation/sanitize_values.ts
export class SanitizeValuesMixin {
  static sanitize_string(value: unknown): string {
    if (typeof value !== "string") {
      return String(value);
    }
    return value.replace(/\0/g, "");
  }
}

export function sanitize_input(value: unknown): unknown {
  if (typeof value === "string") {
    return SanitizeValuesMixin.sanitize_string(value);
  }
  return value;
}

File: skypydb/security/internal/validation/sanitize_values.py
"""Sanitization helpers used by input validation mixins."""

from typing import Any


class SanitizeValuesMixin:
    """Provide basic sanitization for user-supplied values."""

    @staticmethod
    def sanitize_string(value: str) -> str:
        """Normalize strings by removing null bytes and coercing non-strings."""

        if not isinstance(value, str):
            return str(value)
        return value.replace("\x00", "")

def sanitize_input(value: Any) -> Any:
    """Sanitize input values using the same behavior as `SanitizeValuesMixin`."""

    if isinstance(value, str):
        return SanitizeValuesMixin.sanitize_string(value)
    return value
